tomviz/RecentFilesMenu.cxx

```diff
@@ -285,19 +281,19 @@ void RecentFilesMenu::stateTriggered()
   QAction* actn = qobject_cast<QAction*>(this->sender());
   Q_ASSERT(actn);
 
-  if (!QFileInfo::exists(actn->iconText())) {
-    // This should never happen since the checks in aboutToShowMenu should
-    // prevent it, but just in case...
+  if (QFileInfo::exists(actn->iconText())) {
+    QString filename = actn->data().toString();
+    if (SaveLoadStateReaction::loadState(filename)) {
+      // the above call will ensure that the file name moves to top of the list
+      // since it calls pushStateFile() on success.
+      return;
+    }
+  } else {
+    // // If the user tried to open a recent file that no longer exists, remove
+    // it from the recent files
     qWarning() << "Error: file '" << actn->iconText() << "' does not exist.";
-    return;
   }
 
-  QString filename = actn->data().toString();
```

Comment:
Removing this causes a compile error below, where filename is used in the conditional but no longer exists.

Reply:
It was moved inside an if statement above, I moved the declaration back out.